app/models/project_step.rb

```diff
@@ -53,6 +59,7 @@ class NoChildrenAllowedError < StandardError; end
 
   before_save :handle_original_date_logic
   before_save :handle_finalized_at
+  before_save :handle_schedule_decendants
```

Comment:
The alternative would be to leave the column blank, and calculate it on the fly every time.